tests/Operation/WatchFunctionalTest.php

```diff
@@ -0,0 +1,340 @@
+<?php
+
+namespace MongoDB\Tests\Operation;
+
+use MongoDB\ChangeStream;
+use MongoDB\Client;
+use MongoDB\Driver\Manager;
+use MongoDB\Driver\ReadPreference;
+use MongoDB\Driver\Server;
+use MongoDB\Driver\Exception\ConnectionTimeoutException;
+use MongoDB\Operation\DatabaseCommand;
+use MongoDB\Operation\InsertOne;
+use MongoDB\Operation\Watch;
+use MongoDB\Tests\CommandObserver;
+use stdClass;
+use ReflectionClass;
+
+class WatchFunctionalTest extends FunctionalTestCase
+{
+    public function setUp()
+    {
+        parent::setUp();
+
+        if ($this->getPrimaryServer()->getType() === Server::TYPE_STANDALONE) {
+            $this->markTestSkipped('$changeStream is not supported on standalone servers');
+        }
+
+        if (version_compare($this->getFeatureCompatibilityVersion(), '3.6', '<')) {
+            $this->markTestSkipped('$changeStream is only supported on FCV 3.6 or higher');
+        }
+    }
+
+    public function testNextResumesAfterCursorNotFound()
+    {
+        $this->insertDocument(['_id' => 1, 'x' => 'foo']);
+
+        $operation = new Watch($this->manager, $this->getDatabaseName(), $this->getCollectionName(), [], ['maxAwaitTimeMS' => 100]);
+        $changeStream = $operation->execute($this->getPrimaryServer());
+
+        $changeStream->rewind();
+        $this->assertNull($changeStream->current());
+
+        $this->insertDocument(['_id' => 2, 'x' => 'bar']);
+
+        $changeStream->next();
+        $this->assertTrue($changeStream->valid());
+
+        $expectedResult = [
+            '_id' => $changeStream->current()->_id,
+            'operationType' => 'insert',
+            'fullDocument' => ['_id' => 2, 'x' => 'bar'],
+            'ns' => ['db' => $this->getDatabaseName(), 'coll' => $this->getCollectionName()],
```

Comment:
Filling in these values from methods makes the test more robust if the PHPUnit configuration changes (that determines the database name). Additionally, I used numeric `_id` values in the documents so we can expect the exact value in the change stream. The change stream's resume token is now the only weak expectation, since we can't predict it.